lib/Doctrine/DBAL/Driver/SQLSrv/SQLSrvConnection.php

```diff
@@ -124,7 +125,19 @@ public function exec($statement)
             throw SQLSrvException::fromSqlSrvErrors();
         }
 
-        return sqlsrv_rows_affected($stmt);
+        $affectedRowCount = sqlsrv_rows_affected($stmt);
+
+        $stmt = sqlsrv_query($this->conn, 'SELECT @@IDENTITY');
```

Comment:
> However I am not sure but I believe those are session related information calls which do not go through the server at all. Would need to investigate this.

This might cause an uproar of DBAs if they actually hit the backend. If only some exotic drivers (*cough* sqlsrv *cough*) actually hit the backend, then we should be quite OK.